agent/engine/daemonmanager/daemon_manager_linux.go

```diff
@@ -111,7 +129,8 @@ func (dm *daemonManager) CreateDaemonTask() (*apitask.Task, error) {
 	// directly, and the object only contains healthcheck, all other fields will be written as empty/nil
 	// in the result string. This will override the configurations that comes with the container image
 	// (CMD for example)
-	rawConfig = fmt.Sprintf("{\"Healthcheck\":%s}", string(rawHealthConfig))
+	rawConfig = fmt.Sprintf("{\"Healthcheck\":%s, \"User\":\"0\"}", string(rawHealthConfig))
```

Comment:
How about using the `daemonUID` constant here instead of hardcoding `0`?

Reply:
the user must be root in order to perform the mount. (I'd previously tested with daemonUID).  will update this with a managed daemon runas user -- that's my miss.